File: TravelFrontend/travel/src/api/site.js
import axios from './axios'

export default {
    showSiteList() {
        return axios.get('/site/list')
    },
    showSiteLike(siteName) {
        return axios.get('/site/getsitelike/' + siteName)
    },
    showSiteNameList() {
        return axios.get('/site/list/sitename')
    },
    showSiteIdBySiteName(siteName) {
        return axios.get('/site/getsiteid/' + siteName)
    },
    showSiteNameBySiteId(siteId) {
        return axios.get('/site/getsitename/' + siteId)
    }
}

File: TravelBackend/travel/src/main/java/com/my/springboot/travel/entity/Site.java
package com.my.springboot.travel.entity;

import com.my.springboot.travel.model.SiteDTO;
import lombok.Data;
import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Data
@Entity(name = "site")
public class Site implements Serializable {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Integer siteId;

    @Column
    private String siteName;
    private String siteAddress;

    @Column
    private String sitePhoto;

    @ManyToOne
    @JoinColumn(name = "country_id")
    private Country country;

    @OneToMany(mappedBy = "site")
    private List<Tip> tip;

    public SiteDTO toSiteDTO(){
        SiteDTO siteDTO = new SiteDTO();
        siteDTO.setSiteId(this.getSiteId());
        siteDTO.setSiteName(this.getSiteName());
        siteDTO.setSiteAddress(this.getSiteAddress());
        siteDTO.setSitePhoto(this.getSitePhoto());
        return siteDTO;
    }


}
